frame/example/src/lib.rs

```diff
@@ -598,6 +598,14 @@ pub mod pallet {
 	#[pallet::getter(fn foo)]
 	pub(super) type Foo<T: Config> = StorageValue<_, T::Balance, ValueQuery>;
 
+	#[pallet::storage]
+	pub type MyMap<T> = StorageMap<_, Blake2_128Concat, u8, u16>;
+
+	#[pallet::storage]
+	pub type MyCounter<T> = StorageValue<_, u32, ValueQuery>;
+
+	pub type CountedMap<T> = CountedStorageMap<u8, u16, MyMap<T>, MyCounter<T>>;
```

Comment:
While I see what you're going for here, this is a kind of cumbersome declaration style. Ideally we could make it look like a single storage item:

```rust
#[pallet::storage]
pub type CountedMap<T> = CountedStorageMap<u8, u16>;
```

Reply:
I'll try to implement it as you suggest, WIP

Reply:
should be done now, I still want to make a final review myself, but it is ready for review